src/AppInstallerRepositoryCore/Microsoft/SQLiteIndex.cpp

```diff
@@ -35,49 +36,53 @@ namespace AppInstaller::Repository::Microsoft
         return result;
     }
 
-    std::unique_ptr<Schema::ISQLiteIndex> SQLiteIndex::CreateISQLiteIndex() const
+    std::unique_ptr<Schema::ISQLiteIndex> SQLiteIndex::CreateISQLiteIndex(const Schema::Version& version)
     {
         using namespace Schema;
 
-        if (m_version == Version{ 1, 0 })
+        if (version == Version{ 1, 0 })
         {
             return std::make_unique<V1_0::Interface>();
         }
-        else if (m_version == Version{ 1, 1 })
+        else if (version == Version{ 1, 1 })
         {
             return std::make_unique<V1_1::Interface>();
         }
-        else if (m_version == Version{ 1, 2 })
+        else if (version == Version{ 1, 2 })
         {
             return std::make_unique<V1_2::Interface>();
         }
-        else if (m_version == Version{ 1, 3 })
+        else if (version == Version{ 1, 3 })
         {
             return std::make_unique<V1_3::Interface>();
         }
-        else if (m_version == Version{ 1, 4 })
+        else if (version == Version{ 1, 4 })
         {
             return std::make_unique<V1_4::Interface>();
         }
-        else if (m_version == Version{ 1, 5 })
+        else if (version == Version{ 1, 5 })
         {
             return std::make_unique<V1_5::Interface>();
         }
-        else if (m_version == Version{ 1, 6 } ||
-            m_version.MajorVersion == 1 ||
-            m_version.IsLatest())
+        else if (version == Version{ 1, 6 })

```

Comment:
Should these be in reverse order to optimize the comparisons? I assume the newer schema versions would be more common, so having them at the top of the comparisons would result in slightly fewer comparisons

Reply:
I did one better.

Reply:
That you did! Much cleaner!